### DISCUSSION TITLE
Why is SuiteSparse built as shared and static library?

### DISCUSSION THREAD

**Discussion by jhacsonmeza** [DrTimothyAldenDavis/SuiteSparse#873] [Q&A] [answered]
Sorry for this naive question, but I don't understand why SuiteSparse is built by default as a shared and static library. I'm building SuiteSparse to compile another library. I just want to understand if for my case it's just enough to build SuiteSparse only as shared library.

**Comment by mmuetzel** [comment, marked as answer]
Afaict, shared *and* static libraries are built for (most of) SuiteSparse to make life easier for distributors (like Debian, Homebrew, ...). That way they can distribute binary packages that allow developers using these packages to decide whether they prefer shared or static linking.

If you already know that you'll only need shared libraries and you are building SuiteSparse yourself, you can disable building static libraries.

**Reply by jhacsonmeza** [reply]
I understand. Thank you for the clarification.
